### PR TITLE
test: Add Cloud Firewall for integration tests

### PR DESCRIPTION
**Pull request by ykim-1**
## 📝 Description

This PR implements Linode Cloud Firewall for integration tests to enhance security.

Note: GHA does not support ipv6 so only ipv4 will get added in firewall during GHA execution. However, ipv6 will get added automatically if ipv6 address and route exist

## ✔️ How to Test

Still creating new fixtures for some updated tests 

## 📷 Preview

**If applicable, include a screenshot or code snippet of this change. Otherwise, please remove this section.**

### CODE REVIEW

**Comment by zliang-akamai** [comment]
Can we take this chance to upgrade g5 Linodes to g6?

**Comment by ykim-1** [comment]
> Can we take this chance to upgrade g5 Linodes to g6?

All the tests are using g6 instances 👍

File: test/integration/example_integration_test.go [hunk 1]
```diff
@@ -81,7 +81,7 @@ func Example() {
 	fmt.Println("### Error:", err)
 
 	if spendMoney {
-		linode, err = linodeClient.CreateInstance(context.Background(), linodego.InstanceCreateOptions{Region: "us-southeast", Type: "g5-nanode-1"})
+		linode, err = linodeClient.CreateInstance(context.Background(), linodego.InstanceCreateOptions{Region: "us-southeast", Type: "g5-nanode-1", FirewallID: GetFirewallID()})
```
Comment (zliang-akamai) [inline test/integration/example_integration_test.go:84]: This seems to be the last g5 instance in our test. Can we upgrade it?

**Review by zliang-akamai — APPROVED** [review]
Nice work!

**Review by jriddle-linode — APPROVED** [review]
LGTM, works locally

**Pull request merged**